### ISSUE TITLE
[MacOS] Freeze after 20+ minutes of running

### ISSUE THREAD

**Issue by EdenSG** [transmission/transmission#106]
After about 20 minutes of running (this varies), Transmission will freeze (spinning beachball). No information is logged, and it doesn't crash.

When started from a terminal (`/Applications/Transmission.app/Contents/MacOS/Transmission`), the line `[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0` shows up, but then seems to run normally until it freezes again.

MacOS 10.12.1 (16B2657)
Transmission 2.92 (14714)

**Comment by sevenfourk** [comment]
Hey @EdenSG, "freeze" happened to me very randomly. I ended up using it like in old-linux-days, running a deamon, and using web UI. On Linux I used gtk+ version, but this macOS gui does not satisfy me at all. Been trying out it randomly couple of times in a year, the same situation, freezes and hangs.

Just wanted to confirm that I experienced that too.

**Comment by mikedld** [comment]
Did you try the [nightly build](https://build.transmissionbt.com/job/trunk-mac/lastSuccessfulBuild/)? Is Transmission window in background (inactive) or in foreground (active, top-most), or hidden, or minimized all that time? What do you have sort order set to? What happens if you change the sort order to something else?

Are there hang reports in Console? If not, try running `sample Transmission` in Terminal when this happens please. In either case, attach the report here.

**Comment by EdenSG** [comment]
@mikedld No, I haven't tried the nightly, and it was always in the background (either hidden, minimized, or behind other windows). Sort order is by date added. I haven't had the issue in the last two weeks as far as I can remember though – I did a MacOS update (to 10.12.2) and it seemed to stop happening. I'll `sample Transmission` in the future if it does happen again.

**Comment by metaclam** [comment]
Having a very similar issue with 2.9.4, when I hadn't with 2.9.3.  I posted in forums about this. I don't know whether it is because:

• update from 2.9.3 > 2.9.4
• update MacOS to 10.13.4
• fact that I mistakenly paused 3700+ torrents and then had to restart them all*

* not sure if days later it is still "catching up" to restarting them.  log shows a lot of saving of .resume files. But they appear to be in batches.  

I've got a number of crash logs, but they only seem to be generated when I kill -3 transmission in terminal.  (I've learned that only killing it allows it to relaunched;  force quitting won't allow relaunch without system restart.)

**Comment by ckerr** [comment]
I'm closing this issue due to its age and I can't reproduce it.

If this is still an issue in the [nightly builds](https://transmissionbt.org/download) please file a new bug with updated information. Thank you!
